### PR TITLE
DD-700: Exclude easy-migration metadata files from embargo

### PR DESCRIPTION
**Pull request by vesaakerman**
Fixes DD-700

# Description of changes
files in `easy-migration` don't come under `embargo`

# How to test

# Related PRs 
(Add links)
* 

# Notify
@DANS-KNAW/dataversedans

### CODE REVIEW

File: src/main/scala/nl.knaw.dans.easy.dd2d/DatasetEditor.scala [hunk 1]
```diff
@@ -133,11 +133,12 @@ abstract class DatasetEditor(instance: DataverseInstance, optFileExclusionPatter
     } yield ()
   }
 
-  protected def getAllFiles(persistendId: PersistendId): Try[List[FileMeta]] = {
+  protected def getFilesToEmbargo(persistendId: PersistendId): Try[List[FileMeta]] = {
     for {
       r <- instance.dataset(persistendId).listFiles()
       files <- r.data
-    } yield  files
+      filesToEmbargo = files.filter(f => f.directoryLabel.getOrElse(true).toString != "easy-migration")
```
Comment (janvanmansum) [inline src/main/scala/nl.knaw.dans.easy.dd2d/DatasetEditor.scala:140]: If there is no directoryLabel you will end up comparing "true" != "easy-migration" which doesn't make any sense.

I think you need something like:

f.directoryLabel.map(_ != "easy-migration").getOrElse(true)

File: src/main/scala/nl.knaw.dans.easy.dd2d/DatasetUpdater.scala [hunk 2]
```diff
@@ -97,8 +97,8 @@ class DatasetUpdater(deposit: Deposit,
 
           dateAvailable <- deposit.getDateAvailable
           _ <- if (isEmbargo(dateAvailable)) {
-            val fileIdsToEmbargo = fileReplacements.keys ++ fileAdditions.keys
-            logger.info(s"Embargoing new files until ${ dateAvailable }")
+            val fileIdsToEmbargo = (fileReplacements ++ fileAdditions).filter(f => f._2.directoryLabel.getOrElse(true).toString != "easy-migration").keys
```
Comment (janvanmansum) [inline src/main/scala/nl.knaw.dans.easy.dd2d/DatasetUpdater.scala:100]: Same here.

**Review by janvanmansum — CHANGES_REQUESTED** [review]
You've found the right places in the code to edit, but there are still some problems. See my comments.

Please, also make sure you do a test with an update-deposit with an embargo as well.

File: src/main/scala/nl.knaw.dans.easy.dd2d/DatasetEditor.scala (hunk 1 above)
Comment (vesaakerman) [reply src/main/scala/nl.knaw.dans.easy.dd2d/DatasetEditor.scala]: Autch, it had to of course be  a string....

**Pull request merged**